### PR TITLE
Implement `STRJOIN` to join strings with a separator

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes bullet point 2 of #625

### CODE REVIEW

**Comment by Rangi42** [comment]
The stack code is in a utility file like the hashmap since it will likely be useful for more functions.

**Comment by Rangi42** [comment]
This also corrects the documentation for `STRCAT`.

**Comment by ISSOtm** [comment]
I'm less sure about this function because I don't see the use case for it.

**Comment by Rangi42** [comment]
Partly it would be for people who use `PRINTT`, `PRINTV`, etc for direct output: this would let them join values with commas, tabs, or pipes to output CSV/TSV/PSV data.

It would also be useful for the same things as `STRUPR`/`STRLWR`: building `CONSTANT` or `label` names out of other pieces. For instance, `STRJOIN("_", STRUPR(prefix), STRUPR(label), STRUPR(suffix))`.

**Comment by ISSOtm** [comment]
Those use cases would have a fixed number of arguments, so I think they would be a better fit for `STRFMT`. I don't want to complexify the language too much, as parsing is already our main overhead.

**Comment by Rangi42** [comment]
Alright.

**Comment by Rangi42** [comment]
I thought about reopening this feature request, since `STRJOIN` seemed like a way to do ad-hoc fold/reduce operations (e.g. sum, product, etc). However, it would have too many problems:

- A zero-arg default value would have to be given, e.g. `STRJOIN("+", "0", \#)` (and then it would need trailing comma support).
- Operator precedence would require each arg to be parenthesized, so you'd actually have to do `STRCAT("(", STRJOIN(")+(", "0", \#), ")")`.
- The arguments would have to be strings, which is unlikely if you want to join them with `+` or something mathematical.

In short, fold/reduce shouldn't be done by building up code in a string. Just use a `FOR` loop; RGBASM is not a functional language. :P
